### DISCUSSION TITLE
Leading and trailing icon position in button link

### DISCUSSION THREAD

**Discussion by aweell** [Telefonica/mistica-design#643] [Proposals]
Permitir indicar la posición del asset icono en los botones:

![Screenshot 2022-08-02 at 15 30 36](https://user-images.githubusercontent.com/44420072/182386942-c1615ea9-4387-468d-8591-774ed00fb3c6.png)
 
Conecto con https://github.com/Telefonica/mistica-design/issues/631 por si tendría sentido juntar las dos en una release como update de éste componente.

## Implementación

Usar gap en vez del margin para controlar el espaciado entre elementos.

[Tal y como está ahora](https://mistica-ky88s5zeu-tuentisre.vercel.app/playroom#?code=N4Igxg9gJgpiBcIA8BlALgQzAawAQGcAHLGAXmAEYA2AXwD4AdAO11yQCEBXNNCJgBQBOASwC2GQQE9cfITHz5yACgCUuUnVwB3YUygQtAOgwAbGILRKGIQgAsIvAITWV9Zq1YAVDNhi4MuHYOEO4eSACSkAL2vADCGKLmGABKMADmnCYSuJAmEIKk1mCcgoIwTGixEHmC1rgA9IwsbPVcPLIi4lJNoRzcvAKdEtId8orAquqaOnoGxmYWVjYxEM4grk0ebJGyK-GJginpmdm5%2BYXgJWUVVTV1jaFePn4BQby9rf0dYsNNSPXoLDYJogAA0IDQthgiXwCAA2iAALIQABuwnwmEEIAAuuCdFBIbD4HCAMxUAAM2JoQA) puede cambiarse la posición del asset en función de la estructura, pero no se respeta el espaciado correctamente:

**Comment by yceballost** [comment]
Qué casos crees que puede tener sentido para el trailing? En el caso del buttonLink es evidente que es para el chevron que plantea Movistar

**Reply by aweell** [reply]
Es un buen punto, habría que echarle un ojo a eso (Estaba contemplando el button como una única implementación `<Button>`) Pero son componentes separados...

Se me ocurre en primera instancia que se quieran usar botones como trigger de un dropdown menu:

![Screenshot 2022-08-02 at 15 22 45](https://user-images.githubusercontent.com/44420072/182385260-1fbf330f-9035-4837-ab24-46de44bc3c7c.png)

Así de paso saco a la palestra otro tema que quería tocar con respecto al trigger del [componente menu](https://mistica-web.vercel.app/?path=/story/components-menu--default).

https://media.giphy.com/media/Hj7SSRD2lAeQ0/giphy.gif

**Reply by yceballost** [reply]
Yo si de momento no existe una necesidad real no lo metería, lo dejaría on hold.

**Reply by aweell** [reply]
Cambio el scope de la discussion únicamente a `buttonLink`

**Reply by yceballost** [reply]
Lo que se queda como discussion entonces es por un lado cómo nombramos a ese trailing / leading? y la implementación?

**Reply by aweell** [reply]
Se quedaría:

1. Si tiene sentido que existan dos posiciones para el icono en el  `<ButtonLink>` ya que en el ejemplo de #631 sólo se muestra una.
2. En el caso de que lo tenga, si `leading` o `trailing` son el nombre apropiado para indicar estas posiciones (También es habitual encontrarlo como `start`/`end` o `before`/`after`)
3. (Dejaría los detalles de implementación para la definición de las specs)

**Comment by yceballost** [comment]
Finalmente vamos con la opción `Leading` y `Trailing` para la nomenclatura de la posición del icono.
